server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/api/MailQueueView.java

```diff
@@ -38,6 +38,8 @@
 
     CompletableFuture<Long> delete(DeleteCondition deleteCondition);
 
+    CompletableFuture<Void> delete(Mail mail);
```

Comment:
I don't agree with the solution: we should not have to worry about which API to call for a specific case. Here, we can have the mail deleted with 2 different APIs and it's not good. Would you mind handling the specific case of DeleteCondition.withName in existing method to forward it to the method you just created (and change it to private) ?